File: Exp03-Notepad/app/src/main/java/com/example/notepad/MainActivity.java
package com.example.notepad;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.content.Intent;
import android.os.Bundle;

import com.example.notepad.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

  private ActivityMainBinding _l;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    _l = ActivityMainBinding.inflate(getLayoutInflater());
    setContentView(_l.getRoot());

    RecyclerView view = _l.view;
    NoteHelper helper = new NoteHelper(this, "notes", null, 1);
    NotesAdapter adapter = new NotesAdapter(helper.select(null, null));
    view.setAdapter(adapter);
    StaggeredGridLayoutManager manager = new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL);
    view.setLayoutManager(manager);

    _l.addNote.setOnClickListener(v -> {
      Intent intent = new Intent(this, NewNoteActivity.class);
      startActivity(intent);
    });
  }

}

File: Exp04-CloudMusic/app/src/main/java/com/example/cloudmusic/MainActivity.kt
package com.example.cloudmusic

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.cloudmusic.adapters.BottomNavAdapter
import com.example.cloudmusic.databinding.ActivityMainBinding
import com.example.cloudmusic.fragments.NavFindFragment
import com.example.cloudmusic.fragments.NavPersonFragment

class MainActivity : AppCompatActivity() {
  private lateinit var binding: ActivityMainBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = ActivityMainBinding.inflate(layoutInflater)
    setContentView(binding.root)

    if (savedInstanceState == null) {
      createViewPage2()
    }
  }

  private fun createViewPage2() {
    binding.viewPage2.adapter = BottomNavAdapter(
      this, listOf(NavFindFragment(), NavPersonFragment())
    )

    binding.navHome.setOnClickListener {
      binding.viewPage2.currentItem = 0
    }

    binding.navMy.setOnClickListener {
      binding.viewPage2.currentItem = 1
    }
  }
}